OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -96,18 +110,22 @@ protected virtual bool AllowIdleAnimation(Actor self)
 			return !IsModifyingSequence;
 		}
 
-		public void Attacking(Actor self, Target target)
+		public void Attacking(Actor self, Target target, Armament a)
 		{
-			if (!string.IsNullOrEmpty(Info.AttackSequence) && DefaultAnimation.HasSequence(NormalizeInfantrySequence(self, Info.AttackSequence)))
+			string sequence;
+			if (!Info.AttackSequences.TryGetValue(a.Info.Name, out sequence))
+				sequence = Info.DefaultAttackSequence;
+
+			if (!string.IsNullOrEmpty(sequence) && DefaultAnimation.HasSequence(NormalizeInfantrySequence(self, sequence)))
 			{
 				state = AnimationState.Attacking;
-				DefaultAnimation.PlayThen(NormalizeInfantrySequence(self, Info.AttackSequence), () => state = AnimationState.Idle);
+				DefaultAnimation.PlayThen(NormalizeInfantrySequence(self, sequence), () => state = AnimationState.Idle);
 			}
 		}
 
 		void INotifyAttack.PreparingAttack(Actor self, Target target, Armament a, Barrel barrel)
 		{
-			Attacking(self, target);
+			Attacking(self, target, a);
```

Comment:
I believe this change triggered a Coverity defect: https://scan8.coverity.com/reports.htm#v24414/p10888/fileInstanceId=13580582&defectInstanceId=4704498&mergedDefectId=153439